vcpkg.json

```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": [
+    "detours",
+    "catch2"
```

Comment:
I'm going to delay doing this for now, but we might want to consider specifying explicit versions so we don't unexpectedly break. E.g. the current version of `catch2` is what they call "v3" which had some breaking changes I needed to fix...

Reply:
I'm a fan of explicit version dependencies, at least to major/minor level.

Reply:
Okay, I _think_ I understand a bit more about how vcpkg handles versioning. From what I can gather, it doesn't really follow the format that something like npm uses and I'm unsure if it's possible to declare a package to use a specific major version (e.g. force using `3.*`). That said, you can specify an explicit baseline - which I've added now - and from my understanding, that should use the package versions listed in that baseline. So, while we're not specifying explicit versions for each package, we should still get consistency and reliability if I'm understanding things correctly.